### PR TITLE
triheatmap

### PR DESCRIPTION
**Pull request by juliettelavoie**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [ ] (If applicable) Tests have been added.
- [x] CHANGELOG.rst has been updated (with summary of main changes).
  - [ ] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* New function to do triangle heatmap

### Does this PR introduce a breaking change?
no

### CODE REVIEW

**Comment by Zeitsperre** [comment]
Ignore the `finish` failure. It's being fixed upstream.

**Comment by juliettelavoie** [comment]
I am thinking about this more and I am not sur if the list of `plot_kw` and `cbar_kw` is really in the figanos style. I am wondering if it should be a dict instead. In that case, should data be also a dict with 2 or 4 element (one ds per triangle type) ? What makes the most sense to you 1 3D dataset or 2/4 2D datasets?

File: src/figanos/matplotlib/plot.py [hunk 1]
```diff
@@ -2768,3 +2769,234 @@ def partition(
...
+        ]
+        plt.xticks(np.array(range(m)), labels=labels_x, rotation=45)
+        plt.yticks(np.array(range(n)), labels=labels_y, rotation=90)
+
+    else:
+        raise ValueError(
+            f"The length of the dimensiondim ({dim},{len(d)}) should be either 2 or 4. It represents the number of triangles."
+        )
+
+    plt.xlabel(other_dims[0])
```
Comment (sarahclaude) [inline src/figanos/matplotlib/plot.py:2960]: Could xlabel or ylabel also be linked to use_attrs?

**Review by sarahclaude — APPROVED** [review]
Could vmin and vmax be automatically added by finding min/max values from da when user don't add them to plot_kw?

Thanks Juliette, looks great!

File: src/figanos/matplotlib/plot.py [hunk 2]
```diff
@@ -2768,3 +2769,234 @@ def partition(
     ax.legend(**legend_kw)
 
     return ax
+
+
+def triheatmap(
+    data: xr.DataArray | xr.Dataset,
+    dim: str,
```
[the remark this answers is not in the document]
Comment (juliettelavoie) [reply src/figanos/matplotlib/plot.py:2776]: well hue means color, I'm not sure it's adapted here has the triangles could be the same colors. how about `tri_dim` or `triangle` ?

File: src/figanos/matplotlib/plot.py [hunk 3]
```diff
@@ -2768,3 +2769,234 @@ def partition(
...
+    """Create a triangle heatmap from a DataArray.
+
+    Note that most of the code comes from:
+    https://stackoverflow.com/questions/66048529/how-to-create-a-heatmap-where-each-cell-is-divided-into-4-triangles
+
+    Parameters
+    ----------
+    data : DataArray or Dataset
+        Input data do plot.
+    ax : matplotlib axis, optional
```
[the remark this answers is not in the document]
Comment (juliettelavoie) [reply src/figanos/matplotlib/plot.py:2795]: oups!

File: src/figanos/matplotlib/plot.py (hunk 1 above)
Comment (juliettelavoie) [reply src/figanos/matplotlib/plot.py]: yes!

**Pull request merged**